packages/server/src/integrations/tests/rest.spec.ts

```diff
@@ -611,4 +633,56 @@ describe("REST Integration", () => {
     expect(calledConfig.headers).toEqual({})
     expect(calledConfig.agent.options.rejectUnauthorized).toBe(false)
   })
+
+  it("uploads file to object store and returns signed URL", async () => {
+    const responseData = Buffer.from("test file content")
+    const filename = "test.tar.gz"
+    const contentType = "application/gzip"
+
+    ;(fetch as unknown as jest.Mock).mockImplementationOnce(() =>
+      Promise.resolve({
+        headers: {
+          raw: () => ({
+            "content-type": [contentType],
+            "content-disposition": [`attachment; filename="${filename}"`],
```

Comment:
Would be good to have a test that also sets a `filename*` that has some non-ascii characters in it to make sure we don't choke on it, and we correctly prioritise the name.